compiler/src/dotty/tools/dotc/typer/RefChecks.scala

```diff
@@ -75,6 +75,21 @@ object RefChecks {
     }
   }
 
+  /** Check that a stable identifier pattern is indeed stable (SLS 8.1.5)
+   */
+  private def checkStableIdentPattern(tree: Tree)(implicit ctx: Context) = {
```

Comment:
I believe it would be simpler to put this check in Typer. 

Here, we could just add `if (ctx.mode.is(Mode.Pattern) && !tree.tpe.isStable) error` to the checks for Ident and Select.

Reply:
But I am also no sure why we should demand stability? What would go wrong if we didn't?

Reply:
The following check seems too strong to me:

```
if (ctx.mode.is(Mode.Pattern) && !tree.tpe.isStable) error
```

As a selection of `unapply` would fail this check. The check should only happen for _Stable Identifier Pattern_ (SLS 8.1.5). I don't know what's the motivation for the Scala specification. Arguably, a `var` could be allowed in a pattern, but then it complicates the specification and check.

Reply:
OK, so let's keep the test for stable idents but move it to Typer.

Reply:
OK, now it's done.